pkg/signup/service/signup_service.go

```diff
@@ -318,7 +318,7 @@ func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, e164PhoneNumber string) er
 		return errors3.NewInternalError(err, "failed listing userSignups")
 	}
 	for _, signup := range userSignupList.Items {
-		if signup.Name != userID {
+		if signup.Spec.UserID != userID && signup.Annotations[v1alpha1.UserSignupUserEmailAnnotationKey] != email {
```

Comment:
Shouldn't we check if the other usersignup (different ID but same email) is deactivated?
We should allow re-using the same phone for different usersignups with the same email if there is no active usersignups with that phone and email.
But if there is already an active usersignups with the same email and phone then we should not allow to re-use the same phone again.
Otherwise it would be possible to register multiple active users with the same email and phone.
Or I'm missing something?

Reply:
Sure, these are the rules I was unclear about.  Also I think we should check that the existing UserSignup has `VerificationRequired` set to false, to confirm that the phone number has  actually been successfully verified already.  Does this make sense?

Reply:
New signup == new verification.

Reply:
It's a bit challenging to visualise all the possible combinations, so let's spell them out here - let's assume that for all conditions, the phone number is the same:

1. Different user ID, Same email, Not Deactivated - (Block)
2. Different user ID, Same email, Deactivated (Allow)
3. Different user ID, Different email, Not Deactivated (?)
4. Different user ID, Different email, Deactivated (Allow?)

Reply:
3. Block
4. I would allow if it's easy to implement (not critical though)

Reply:
So it seems that we don't require an email comparison at all then - just block if the phone number is tied to a non-deactivated UserSignup with the same phone number, otherwise don't block?

Reply:
Yes, looks like that. But we should be careful with that. It should be impossible to register multiple accounts with the same phone first (UserSignups created, verification code generated and sent) and then verify them all.